### PR TITLE
feat(team-roles): Disable deprecated roles in OrgRoleSelect

### PR DESCRIPTION
**Pull request by leedongwei**
Screen Shot 2022-06-05 at 3 29 32 PM

### CODE REVIEW

File: static/app/views/settings/organizationMembers/organizationMemberDetail.tsx [hunk 1]
```diff
@@ -52,8 +52,6 @@ type Props = {
 
 type State = {
   member: Member | null;
-  roleList: Member['roles'];
-  selectedRole: Member['role'];
```
Comment (leedongwei) [inline static/app/views/settings/organizationMembers/organizationMemberDetail.tsx:56]: These were never used at all.

File: static/app/views/settings/organizationMembers/inviteMember/orgRoleSelect.tsx [hunk 2]
```diff
@@ -17,33 +17,37 @@ const Label = styled('label')`
...
+    const {disabled, enforceRetired, enforceAllowed, roleList, roleSelected} = this.props;
 
     return (
       <Panel>
-        <PanelHeader>{t('Role')}</PanelHeader>
+        <PanelHeader>{t('Organization Role')}</PanelHeader>
 
         <PanelBody>
           {roleList.map(role => {
             const {desc, name, id, allowed} = role;
```
Comment (nhsiehgit) [inline static/app/views/settings/organizationMembers/inviteMember/orgRoleSelect.tsx:36]: should we deconstruct `isRetired` here as well?

**Review by nhsiehgit — APPROVED** [review]
👍 with slight nits

File: src/sentry/api/serializers/models/organization_member/expand/roles.py [hunk 3]
```diff
@@ -15,11 +14,17 @@
 
 
 def _is_retired_role_hidden(role: OrganizationRole, member: OrganizationMember) -> bool:
+    """
+    During EA, we will show the role but make it greyed out and prevent the user
```
Comment (nhsiehgit) [inline src/sentry/api/serializers/models/organization_member/expand/roles.py:18]: Does this helper determine whether it's hidden or greyed out?

File: static/app/views/settings/organizationMembers/inviteMember/orgRoleSelect.tsx [hunk 4]
```diff
@@ -17,33 +17,37 @@ const Label = styled('label')`
...
+            const {desc, name, id, allowed, isRetired: roleRetired} = role;
+
+            const isRetired = enforceRetired && roleRetired;
+            const isDisabled = disabled || isRetired || (enforceAllowed && !allowed);
+
             return (
               <PanelItem
                 key={id}
-                onClick={() => !isDisabled && this.props.setRole(id)}
+                onClick={() => !isDisabled && this.props.setSelected(id)}
```
Comment (nhsiehgit) [inline static/app/views/settings/organizationMembers/inviteMember/orgRoleSelect.tsx:44]: can we deconstruct `setSelected` from the props above too?

**Review by nhsiehgit — APPROVED** [review]
more nits, non blocking

File: src/sentry/api/serializers/models/organization_member/expand/roles.py (hunk 3 above)
Comment (leedongwei) [reply src/sentry/api/serializers/models/organization_member/expand/roles.py:18]: Yea it doesn't pass the role to the frontend, so it doesn't even appear at all.

**Pull request merged**